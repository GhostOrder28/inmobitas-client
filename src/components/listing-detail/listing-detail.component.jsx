import React from 'react';
import { strParseOut } from '../../utils/utility-functions';
import {
  StyledTable,
  StyledTbody,
  StyledTr,
  StyledTdWrapped,
  StyledTdUnwrapped,
} from '../../global-styles/table.styles'
import ImageGallery from 'react-image-gallery';
import { Pane } from 'evergreen-ui';
import { ownerLabels, listingLabels, globalLabels } from '../../constants/language/english/english-labels.constants';
import "react-image-gallery/styles/css/image-gallery.css";
import './listing-detail-gallery.styles.css';

import ownersData from '../../assets/data/owners.data'; // TODO: this should be fetched

const ListingDetail = ({ data, photosUrls, table }) => {

  const getOwner = () => {
    return ownersData.find(owner => owner.id === data.ownerId)
  };

  return (
    <div>
      {
        photosUrls.length
          ? <ImageGallery
            items={
              photosUrls.map(photo => ({
                  original: `${photo}`,
                  thumbnail: `${photo}`
              }))
            }
            thumbnailPosition={'bottom'}
            />
          : 'There is no photos...'
      }
      <Pane margin={10}>
        {table}
      </Pane>
    </div>
  )
};

export default ListingDetail;
