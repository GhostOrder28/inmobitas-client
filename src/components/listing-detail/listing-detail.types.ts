export type Picture = {
  filename: string;
  pictureId: number;
  suffix: string;
}
