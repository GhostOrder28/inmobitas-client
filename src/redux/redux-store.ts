import { createStore, applyMiddleware, Middleware, compose } from "redux";
import { persistStore } from "redux-persist";
import logger from "redux-logger";
import rootReducer from "./root-reducer";
import createSagaMiddleware from "redux-saga";
import rootSaga from "./root-saga";

const sagaMiddleware = createSagaMiddleware();

const middlewares: Middleware[] = [sagaMiddleware];

if (process.env.NODE_ENV === "development") {
  middlewares.push(logger);
}

// @ts-ignore
const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;;

const store = createStore(rootReducer, composeEnhancer(applyMiddleware(...middlewares)));
// @ts-ignore
window.store = store
sagaMiddleware.run(rootSaga)
const persistor = persistStore(store);
export { store, persistor };
