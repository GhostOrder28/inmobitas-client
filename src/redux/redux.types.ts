import { 
  InvalidIdentifierErrorData,
  DuplicateEntityErrorData,
  ValidationErrorData,
  ServerErrorData,
} from "../http/http.types";
import { Expand } from "../utils/utility-types";

export type RequestUserInfoFailureError = InvalidIdentifierErrorData | ServerErrorData; // this is not redundant bc there can be other error types when requesting user info
export type SignUpFailureError =  ValidationErrorData | DuplicateEntityErrorData | ServerErrorData;
export type SignInFailureError = Expand<ValidationErrorData | InvalidIdentifierErrorData | ServerErrorData>;
export type GenerateGuestUserError = SignInFailureError | ServerErrorData;
