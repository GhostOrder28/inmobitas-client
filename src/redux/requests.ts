import http from '../utils/axios-instance';
import { AxiosResponse, AxiosError } from 'axios';

export const fetchUserInfo = async (email: string, password: string): Promise<AxiosResponse | AxiosError> => {
  try {
    return await http.post('/signin', { email, password })
  } catch (err) {
    return err as AxiosError
  }
}

export const fetchNewUser = async (names: string, email: string, password: string, confirmPassword: string): Promise<AxiosResponse | AxiosError> => {
  try {
    return await http.post('/signup', { names, email, password, confirmPassword })
  } catch (err) {
    return err as AxiosError
  }
}

