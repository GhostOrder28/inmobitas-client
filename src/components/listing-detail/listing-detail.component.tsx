import React, { useState, useEffect } from "react";
import http from "../../utils/axios-instance";
import { useSelector } from "react-redux";
import { selectCurrentUserId } from "../../redux/user/user.selectors";
import { useParams } from "react-router-dom";
import {
  StyledTable,
  StyledTbody,
  StyledTr,
  StyledTdWrapped,
  StyledTdUnwrapped,
  StyledThWithSpan,
} from "../../global-styles/table.styles";
import { Pane, Spinner } from "evergreen-ui";
import {
  ownerLabels,
  contractLabels,
  estateLabels,
  locationLabels,
  featuresLabels,
  ownerPreferencesLabels,
} from "../../constants/language/english/english-labels.constants";
import { presetSelector } from "../../utils/utility-functions";
import { Tablist, Tab } from "evergreen-ui";
import PhotoGallery from "../photo-gallery/photo-gallery.component";
import "react-image-gallery/styles/css/image-gallery.css";
import "./listing-detail-gallery.styles.css";

import { Listing, Presets } from "../../pages/listing-page/listing-page.types";
import { Picture } from "./listing-detail.types";

type ListingDetailProps = {
  dataPresets: Presets | undefined;
  listing: Listing | undefined;
};

const ListingDetail = ({ dataPresets, listing }: ListingDetailProps) => {
  const userId = useSelector(selectCurrentUserId);
  const { listingid } = useParams();
  const [isHidden, setIsHidden] = useState({
    owner: false,
    contract: false,
    ownerPreferences: false,
    location: false,
    estate: false,
  });
  const [selectedTab, setSelectedTab] = useState(0);
  const [listingPictures, setListingPictures] = useState<Picture[]>([]);

  useEffect(() => {
    (async function () {
      const estatePictures = await http.get<Picture[]>(
        `/estatepictures/${userId}/${listingid}`
      );
      setListingPictures(estatePictures.data);
    })();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return !(listing && dataPresets) ? (
    <Spinner />
  ) : (
    <>
      <Tablist width={"100%"} display={"flex"} className="tablist">
        {["Information", "Gallery"].map((tab, index) => (
          <Tab
            key={tab}
            id={tab}
            onSelect={() => setSelectedTab(index)}
            isSelected={index === selectedTab}
            aria-controls={`panel-${tab}`}
            flex={1}
            height={"2.5rem"}
            borderRadius={0}
          >
            {tab}
          </Tab>
        ))}
      </Tablist>
      <Pane display={selectedTab === 0 ? "block" : "none"} margin={10}>
        <StyledTable>
          <StyledTbody>
            <StyledTr
              onClick={() => {
                setIsHidden({ ...isHidden, owner: !isHidden.owner })
              }}
            >
              <StyledThWithSpan spanValue="2">
                {ownerLabels.OWNER_INFO}
              </StyledThWithSpan>
            </StyledTr>
            {!isHidden.owner && (
              <>
                <StyledTr>
                  <StyledTdUnwrapped>{ownerLabels.OWNER}</StyledTdUnwrapped>
                  <StyledTdWrapped>{listing.clientName}</StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {ownerLabels.OWNER_CONTACT_PHONE}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>
                    {listing.clientContactPhone}
                  </StyledTdWrapped>
                </StyledTr>
              </>
            )}
          </StyledTbody>
          <StyledTbody>
            <StyledTr
              onClick={() => {
                setIsHidden({ ...isHidden, contract: !isHidden.contract });
              }}
            >
              <StyledThWithSpan spanValue="2">
                {contractLabels.CONTRACT}
              </StyledThWithSpan>
            </StyledTr>
            {!isHidden.contract && (
              <>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {contractLabels.CONTRACT_TYPE}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>
                    {presetSelector(
                      dataPresets.contractTypes,
                      listing.contractTypeId,
                      "contract"
                    )}
                  </StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdWrapped>
                    {listing.contractType === 1
                      ? contractLabels.ESTATE_PRICE
                      : contractLabels.RENT}
                  </StyledTdWrapped>
                  <StyledTdWrapped>
                    {listing.estatePrice &&
                      `${presetSelector(
                        dataPresets.currencyTypes,
                        listing.currencyTypeId,
                        "currency"
                      )} ${listing.estatePrice}`}
                  </StyledTdWrapped>
                </StyledTr>
                {listing.utilitiesIncluded !== null && (
                  <StyledTr>
                    <StyledTdUnwrapped>
                      {contractLabels.UTILITIES}
                    </StyledTdUnwrapped>
                    <StyledTdWrapped>
                      {listing.utilitiesIncluded ? "Yes" : "No"}
                    </StyledTdWrapped>
                  </StyledTr>
                )}
                <StyledTr>
                  <StyledTdWrapped>{contractLabels.FEE}</StyledTdWrapped>
                  <StyledTdWrapped>
                    {listing.fee &&
                      `
                      ${
                        !listing.isPercentage
                          ? presetSelector(
                              dataPresets.currencyTypes,
                              listing.currencyTypeId,
                              "currency"
                            )
                          : ""
                      } 
                      ${listing.fee} 
                      ${listing.isPercentage ? "%" : ""}
                    `}
                  </StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdWrapped>
                    {contractLabels.SIGNED_DATE}
                  </StyledTdWrapped>
                  <StyledTdWrapped>{listing.signedDate}</StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdWrapped>{contractLabels.START_DATE}</StyledTdWrapped>
                  <StyledTdWrapped>{listing.startDate}</StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdWrapped>{contractLabels.END_DATE}</StyledTdWrapped>
                  <StyledTdWrapped>{listing.endDate}</StyledTdWrapped>
                </StyledTr>
              </>
            )}
          </StyledTbody>
          {listing.contractType === 2 && (
            <StyledTbody>
              <StyledTr
                onClick={() => {
                  setIsHidden({
                    ...isHidden,
                    ownerPreferences: !isHidden.ownerPreferences,
                  });
                }}
              >
                <StyledThWithSpan spanValue="2">
                  {ownerPreferencesLabels.OWNER_PREFERENCES}
                </StyledThWithSpan>
              </StyledTr>
              {!isHidden.ownerPreferences && (
                <>
                  <StyledTr>
                    <StyledTdUnwrapped>
                      {ownerPreferencesLabels.PETS_ALLOWED}
                    </StyledTdUnwrapped>
                    <StyledTdWrapped>
                      {listing.petsAllowed !== null &&
                        (listing.petsAllowed ? "Yes" : "No")}
                    </StyledTdWrapped>
                  </StyledTr>
                  <StyledTr>
                    <StyledTdUnwrapped>
                      {ownerPreferencesLabels.CHILDREN_ALLOWED}
                    </StyledTdUnwrapped>
                    <StyledTdWrapped>
                      {listing.childrenAllowed !== null &&
                        (listing.childrenAllowed ? "Yes" : "No")}
                    </StyledTdWrapped>
                  </StyledTr>
                  <StyledTr>
                    <StyledTdUnwrapped>
                      {ownerPreferencesLabels.PREFERENCE_DETAILS}
                    </StyledTdUnwrapped>
                    <StyledTdWrapped>
                      {listing.ownerPreferencesDetails || ""}
                    </StyledTdWrapped>
                  </StyledTr>
                </>
              )}
            </StyledTbody>
          )}
          <StyledTbody>
            <StyledTr
              onClick={() => {
                setIsHidden({ ...isHidden, location: !isHidden.location });
              }}
            >
              <StyledThWithSpan spanValue="2">
                {locationLabels.LOCATION}
              </StyledThWithSpan>
            </StyledTr>
            {!isHidden.location && (
              <>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {locationLabels.DISTRICT}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>{listing.district}</StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {locationLabels.NEIGHBORHOOD}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>{listing.neighborhood}</StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {locationLabels.ADDRESS_DETAILS}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>
                    {listing.addressDetails || ""}
                  </StyledTdWrapped>
                </StyledTr>
              </>
            )}
          </StyledTbody>
          <StyledTbody>
            <StyledTr
              onClick={() => {
                setIsHidden({ ...isHidden, estate: !isHidden.estate });
              }}
            >
              <StyledThWithSpan spanValue="2">
                {estateLabels.ESTATE}
              </StyledThWithSpan>
            </StyledTr>
            {!isHidden.estate && (
              <>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {estateLabels.ESTATE_TYPE}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>
                    {presetSelector(
                      dataPresets.estateTypes,
                      listing.estateTypeId,
                      "estate"
                    )}
                  </StyledTdWrapped>
                </StyledTr>
                {listing.estateType !== 1 && (
                  <StyledTr>
                    <StyledTdUnwrapped>
                      {estateLabels.FLOOR_LOCATION}
                    </StyledTdUnwrapped>
                    <StyledTdWrapped>
                      {listing.floorLocation || "-"}
                    </StyledTdWrapped>
                  </StyledTr>
                )}
                {listing.estateType === 1 && (
                  <StyledTr>
                    <StyledTdUnwrapped>
                      {estateLabels.NUMBER_OF_FLOORS}
                    </StyledTdUnwrapped>
                    <StyledTdWrapped>
                      {listing.numberOfFloors || "-"}
                    </StyledTdWrapped>
                  </StyledTr>
                )}
                <StyledTr>
                  <StyledTdUnwrapped>
                    {estateLabels.TOTAL_AREA}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>
                    {listing.totalArea && `${listing.totalArea} m²`}
                  </StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {estateLabels.BUILT_AREA}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>
                    {listing.builtArea && `${listing.builtArea} m²`}
                  </StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {featuresLabels.BEDROOMS}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>
                    {listing.numberOfBedrooms || ""}
                  </StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {featuresLabels.BATHROOMS}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>
                    {listing.numberOfBathrooms || ""}
                  </StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {featuresLabels.GARAGES}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>
                    {listing.numberOfGarages || ""}
                  </StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {featuresLabels.KITCHENS}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>{listing.numberOfKitchens}</StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdUnwrapped>
                    {featuresLabels.NATURAL_GAS}
                  </StyledTdUnwrapped>
                  <StyledTdWrapped>
                    {listing.haveNaturalGas !== null &&
                      (listing.haveNaturalGas ? "Yes" : "No")}
                  </StyledTdWrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdUnwrapped spanValue="2">
                    {estateLabels.ESTATE_DETAILS}
                  </StyledTdUnwrapped>
                </StyledTr>
                <StyledTr>
                  <StyledTdWrapped spanValue="2">
                    {listing.estateDetails || ""}
                  </StyledTdWrapped>
                </StyledTr>
              </>
            )}
          </StyledTbody>
        </StyledTable>
      </Pane>
      <PhotoGallery
        display={selectedTab === 1 ? "block" : "none"}
        listingPictures={listingPictures}
      />
    </>
  );
};
export default ListingDetail;
