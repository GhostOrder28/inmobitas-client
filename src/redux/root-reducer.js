import { combineReducers } from 'redux';
// import { persistReducer } from 'redux-persist';
import storageSession from 'redux-persist/lib/storage/session';

import userReducer from './user/user.reducer';
import testingReducer from './testing/testing.reducer';

// const persistConfig = {
//   key: 'root',
//   storage: storageSession,
// };

const rootReducer = combineReducers({
  user: userReducer,
  test: testingReducer
});

// export default persistReducer(persistConfig, rootReducer);
export default rootReducer;
