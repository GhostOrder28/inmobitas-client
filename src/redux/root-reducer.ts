import { AnyAction, combineReducers } from 'redux';
import { persistReducer, PersistConfig } from 'redux-persist';
import sessionStorage from 'redux-persist/lib/storage/session';
import localStorage from 'redux-persist/lib/storage';
import { UserState } from './user/user.reducer';
import userReducer from './user/user.reducer';

export type RootState = ReturnType<typeof rootReducer>;
type UserReducer = ReturnType<typeof userReducer>;

type ExtendedPersistConfig = PersistConfig<RootState> & {
  blacklist: (keyof RootState)[];
}

type ExtendedUserPersistConfig = PersistConfig<UserState> & {
  blacklist: (keyof UserState)[];
}

const persistConfig = {
  key: 'root',
  storage: localStorage,
  blacklist: ['user']
};

const persistConfigErrors = {
  key: 'user',
  storage: localStorage,
  blacklist: ['errors']
};

const rootReducer = combineReducers({
  user: persistReducer(persistConfigErrors, userReducer),
});

export default persistReducer(persistConfig, rootReducer);
