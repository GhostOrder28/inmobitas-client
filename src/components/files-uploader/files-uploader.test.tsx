import React from 'react';
import FilesUploader from './files-uploader.component';
import { Picture } from '../listing-detail/listing-detail.types';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../redux/redux-store';

it('should render FilesUploader', () => {
  const setStateMock = jest.fn();
  const useStateMock: any = (useState: any) => [useState, setStateMock];
  jest.spyOn(React, 'useState').mockImplementation(useStateMock);
  const [isLoading, setIsLoading] = useStateMock();
  const [noImages, setNoImages] = useStateMock();
  const [files, setFiles] = useStateMock([
    {
      filename: 'myfile.jpg',
      pictureId: 1,
      smallSizeUrl: 'myfile-small.jpg',
      largeSizeUrl: 'myfile-large.jpg'
    }
  ]);

  const snapshot = render(
    <Provider store={store}>
      <FilesUploader
        files={files}
        setFiles={setFiles}
        setIsLoading={setIsLoading}
        setNoImages={setNoImages}
      />
    </Provider>
  );
  expect(snapshot).toMatchSnapshot();
  //screen.debug()
})
