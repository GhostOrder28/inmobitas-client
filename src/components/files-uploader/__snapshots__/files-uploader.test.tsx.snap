// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FilesUploader 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      data-evergreen-toaster-container=""
    >
      <span
        class="css-1sugtjn"
      />
    </div>
    <div>
      <input
        accept="image/*"
        id="upload-btn"
        multiple=""
        type="file"
      />
      <label
        class="upload-btn-label"
        for="upload-btn"
      >
        <svg
          class="css-1oxm3o5 ub-w_20px ub-h_20px ub-box-szg_border-box"
          data-icon="arrow-up"
          viewBox="0 0 20 20"
        >
          <path
            d="M16.7 7.3l-6-6c-.2-.2-.4-.3-.7-.3s-.5.1-.7.3l-6 6c-.2.2-.3.4-.3.7 0 .6.5 1 1 1 .3 0 .5-.1.7-.3L9 4.4V18c0 .5.4 1 1 1s1-.5 1-1V4.4l4.3 4.3c.2.2.4.3.7.3.5 0 1-.4 1-1 0-.3-.1-.5-.3-.7z"
            fill-rule="evenodd"
          />
        </svg>
      </label>
    </div>
  </body>,
  "container": <div>
    <input
      accept="image/*"
      id="upload-btn"
      multiple=""
      type="file"
    />
    <label
      class="upload-btn-label"
      for="upload-btn"
    >
      <svg
        class="css-1oxm3o5 ub-w_20px ub-h_20px ub-box-szg_border-box"
        data-icon="arrow-up"
        viewBox="0 0 20 20"
      >
        <path
          d="M16.7 7.3l-6-6c-.2-.2-.4-.3-.7-.3s-.5.1-.7.3l-6 6c-.2.2-.3.4-.3.7 0 .6.5 1 1 1 .3 0 .5-.1.7-.3L9 4.4V18c0 .5.4 1 1 1s1-.5 1-1V4.4l4.3 4.3c.2.2.4.3.7.3.5 0 1-.4 1-1 0-.3-.1-.5-.3-.7z"
          fill-rule="evenodd"
        />
      </svg>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
